boa/src/value/mod.rs

```diff
@@ -1051,3 +1051,16 @@ impl From<Numeric> for Value {
         }
     }
 }
+
+impl Numeric {
+    #[inline]
+    pub fn not(x: &Self) -> Value {
+        match x {
+            Numeric::Number(x) => {
+                let x = f64_to_int32(*x);
+                Value::from(!x)
+            }
+            Numeric::BigInt(x) => Value::from(JsBigInt::new(!x.as_inner())),
+        }
+    }
+}
```

Comment:
I think it's best if we have separate associative function `not()` one on `JsBigInt` (like we had before) and one on `crate::builtins::Number`.

Reply:
Just wondering why that is, I figured we're only going to call not on the whole Numeric type at this point so surely matching within Numeric makes sense here?

Reply:
> Just wondering why that is, I figured we're only going to call not on the whole Numeric type at this point so surely matching within Numeric makes sense here?

For two reasons:
 - To comply with what the spec is doing, so we can more easily see what is going on.
 - API reasons we would like for the user to able to call `not()` on `JsBigInt` and on `Number` from the rust side.

We shouldn't have `not` on `Numeric` since to call it we it would be messy. `Numeric::not(Numeric::Number(x)).as_number().unwrap()` instead of `Number::not(x)`